packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -111,19 +110,191 @@ export default class {
+
+  private async updateWithoutUpdatingSiblings(
+    graphApi: GraphApi,
+    params: { updatedIndex: number; updatedBy: string },
+  ) {
+    const { updatedIndex, updatedBy } = params;
+
+    const { index: previousIndex } = this;
+
+    if (previousIndex === undefined) {
+      throw new Error("Cannot make an un-ordered link ordered");
+    }
+
+    if (previousIndex === updatedIndex) {
+      throw new Error("No-op: link already has this index");
+    }
+
+    /**
+     * @todo: call dedicated Graph API method to update the index of a link instead of re-creating the link manually
+     * @see https://app.asana.com/0/1202805690238892/1203031430417465/f
+     */
+    await this.removeWithoutUpdatingSiblings(graphApi, {
+      removedBy: updatedBy,
+    });
+
+    const updatedLink = await LinkModel.createLinkWithoutUpdatingSiblings(
+      graphApi,
+      {
+        ...this,
+        index: updatedIndex,
+        createdBy: updatedBy,
+      },
+    );
+
+    return updatedLink;
+  }
+
+  /**
+   * Update the link
+   *
+   * @param params.updatedIndex - the updated index of the link
+   * @param params.updatedBy - the account updating the link
+   */
+  async update(
+    graphApi: GraphApi,
+    params: { updatedIndex: number; updatedBy: string },
+  ) {
+    const { updatedIndex, updatedBy } = params;
+
+    const { index: previousIndex, linkTypeModel } = this;
+
+    const updatedLink = await this.updateWithoutUpdatingSiblings(graphApi, {
+      updatedIndex,
+      updatedBy,
+    });
+
+    const siblingLinks = await this.sourceEntityModel.getOutgoingLinks(
+      graphApi,
+      { linkTypeModel },
+    );
+
+    if (previousIndex === undefined) {
+      throw new Error("Cannot make an un-ordered link ordered");
+    }
+
+    // Whether the index of the link is being increased
+    const isIncreasingIndex = updatedIndex > previousIndex;
+
+    // The minimum index of the affected sibling links
+    const affectedSiblingsMinimumIndex = isIncreasingIndex
+      ? previousIndex + 1
+      : updatedIndex;
+
+    // The maximum index of the affected sibling links
+    const affectedSiblingsMaximumIndex = isIncreasingIndex
+      ? updatedIndex
+      : previousIndex - 1;
+
+    const affectedSiblings = siblingLinks.filter(
+      (sibling) =>
+        sibling.index! >= affectedSiblingsMinimumIndex &&
+        sibling.index! <= affectedSiblingsMaximumIndex,
+    );
+
+    /**
+     * @todo: rely on the Graph API to maintain index integrity of sibling links on updates
+     * @see https://app.asana.com/0/1202805690238892/1203031430417465/f
+     */
+    await Promise.all(
+      affectedSiblings.map((sibling) =>
+        sibling.updateWithoutUpdatingSiblings(graphApi, {
+          updatedIndex: sibling.index! + (isIncreasingIndex ? -1 : 1),
+          updatedBy,
+        }),
+      ),
+    );
+
+    return updatedLink;
+  }
+
+  private async removeWithoutUpdatingSiblings(
+    graphApi: GraphApi,
+    { removedBy }: { removedBy: string },
```

Comment:
Can we use `removedById`

Reply:
https://github.com/hashintel/hash/pull/1081/commits/078568b93b61f7b3a0cb85e506f5f08328b89b0c